streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentInfoTest.java

```diff
@@ -65,10 +65,9 @@ public void shouldDecodePreviousVersion() throws Exception {
         assertEquals(oldVersion.activeTasks, decoded.activeTasks);
         assertEquals(oldVersion.standbyTasks, decoded.standbyTasks);
         assertEquals(0, decoded.partitionsByHostState.size()); // should be empty as wasn't in V1
-        assertEquals(2, decoded.version); // automatically upgraded to v2 on decode;
+        assertEquals(1, decoded.version);
```

Comment:
Is this change intentional? If yes should we rename the test then?

Reply:
I think the test name is still fine -- `AssignmentInfo.decode` should be able to decode a version 1 `Assignment` -- originally, it upgraded the `AssignmentInfo` to be version 2, but I think this is actually not a good idea -- if we receive a version 1 `AssignmentInfo` we should return version 1 and not set version 2.